src/collapse/dtrace.rs

```diff
@@ -126,30 +107,23 @@ impl Folder {
         if let Ok(count) = line.parse::<usize>() {
             self.count = count;
         } else {
-            if let Some((func, addr)) = Folder::stack_line_parts(line) {
-                self.stack.push_front(func.to_string());
+            if let Some((mut funcs, addr)) = Folder::stack_line_parts(line) {
```

Comment:
I think you probably want to call `dejavaify` out here instead, because otherwise you are allocating a `Vec` _on every line_. Take a look at what the [perf code does](https://github.com/jonhoo/inferno/blob/b7605ef69cc53e07612e098264b3bd6499ab5877/src/collapse/perf.rs#L259).

Reply:
Alternatively, you could also pass in `&mut self.stack` to `stack_line_parts`

Reply:
I might misread the perf code or fundamentally misunderstand rust but the way I read the perf code I see one (small) vector per line being allocated in 278 along with a string parsing, then the same string is parsed again in 281.

But if the double allocation is a concern I can optimize it!